openpathsampling/experimental/path_tree/path_tree.py

```diff
@@ -0,0 +1,251 @@
+        trajectory = step.active[self.selector].trajectory
+        return PathTreeStep(
+            offset=0,
+            mccycle=step.mccycle,
+            mover=None,
+            accepted=True,
+            connector=None,
+            trajectory=trajectory,
+            segments=TrajectorySegments(existing=[],
+                                        new=[(0, len(trajectory))]),
+        )
+
+    def _connector(self, shooting_pt, mod_shooting, trial, existing):
+        if shooting_pt is None:
+            return None
+
+        connector_idx = trial.index(mod_shooting)
+        mccycle = existing[shooting_pt]
+        return Connector(connector_idx, mccycle)
+
+    def generate_path_tree_steps(self, steps):
+        steps = iter(self.progress(steps))
+        prev = self._first_path_tree_step(next(steps))
+        path_tree_steps = [prev]
+        connectors = []
+        pre_existing = {snap: prev.mccycle for snap in prev.trajectory}
+        offset = 0
+        for trial, step in self._filter_steps(steps):
+            # track when the snapshot first entered (continuously)
+            current = {snap: step.mccycle for snap in trial}
+            existing = dict(current)
+            existing.update(pre_existing)
+
+            # get information to make the tree step
+            details = step.change.canonical.details
+            shooting_pt, mod_shooting = _shooting_points(details)
+
+            shift = self._get_shift(prev, trial, shooting_pt, mod_shooting)
+            connector = self._connector(shooting_pt, mod_shooting, trial,
+                                        existing)
+
+            if shift is None:
+                offset = 0
+                shift = 0
+
+            tree_step = PathTreeStep(
+                offset=offset + shift,
+                mccycle=step.mccycle,
+                mover=step.change.canonical.mover,
+                accepted=step.change.accepted,
+                trajectory=trial,
+                connector=connector,
+                segments=make_segments(prev.trajectory, trial)
+            )
+
+            if step.change.accepted:
+                offset += shift
+
+            path_tree_steps.append(tree_step)
+
+            if tree_step.accepted:
+                prev = tree_step
+                pre_existing = {k: existing[k] for k in current}
+
+        return path_tree_steps
+
+
+    def svg(self, plotter=None):
+        if plotter is None:
+            plotter = SVGLines
+
+        if type(plotter) is type:
+            plotter = plotter()
+
+        return plotter.draw(self)
+
+    def matplotlib(self, plotter=None):
+        try:
+            import matplotlib.pyplot as plt
+        except ImportError:
+            raise RuntimeError("matplotlib is not installed")
+
+        if plotter is None:
+            plotter = ThinLines
+
+        if type(plotter) is type:
+            plotter = plotter()
+
+        ax = plotter.ax
+        plotter.draw_trajectories(self.path_tree_steps)
```

Comment:
shouldn't this be brought inline with line `234`?
```suggestion
        plotter.draw(self)
```